File: folkmq-lang-js/folkmq-test/demo02-node-ws/_batch_subscribe.js
const {FolkMQ} = require("@noear/folkmq");

async function main() {
    const client =  FolkMQ.createClient("folkmq:ws://127.0.0.1:18602")
        .nameAs("demoapp");

    //订阅
    client.subscribe("test", null, true, message => {
        console.log(message.getBodyAsString());
    });

    client.subscribe("test2", null, true, message => {
        console.log(message.getBodyAsString());
    });

    client.subscribe("test3", null, true, message => {
        console.log(message.getBodyAsString());
    });

    //连接
    await client.connect();

    //发布
    for (let i = 0; i < 10; i++) {
        client.publish("test", FolkMQ.newMessage("test-" + i));
        client.publish("test2", FolkMQ.newMessage("test2-" + i));
    }
}

main();

File: folkmq-lang-python/folkmq/folkmq-test/demo02/_batch_subscribe.py
import asyncio

from socketd.utils.LogConfig import log

from folkmq.FolkMQ import FolkMQ
from folkmq.client.MqMessage import MqMessage


async def main():
    client = (FolkMQ.create_client("folkmq:ws://127.0.0.1:18602")
                    .name_as("demoapp"))

    # 订阅
    await client.subscribe("test", None, True, lambda message:
        log.info(message.get_body_as_string())
    )

    await client.subscribe("test2", None, True, lambda message:
        log.info(message.get_body_as_string())
    )

    await client.subscribe("test3", None, True, lambda message:
        log.info(message.get_body_as_string())
    )

    # 连接
    await client.connect()



    # 发布
    for i in range(10):
        await client.publish("test", MqMessage("test-" + str(i)))
        await client.publish("test2", MqMessage("test2-" + str(i)))

    await asyncio.Future()

if __name__ == "__main__":
    asyncio.run(main())

